docker/test/parsl_mpi_flux_hello.py

```diff
@@ -0,0 +1,87 @@
+#!/bin/env python3
+
+import parsl
+from parsl.app.app import python_app, bash_app
+from parsl.config import Config
+from parsl.channels import LocalChannel
+from parsl.providers import SlurmProvider
+from parsl.executors import FluxExecutor
+from parsl.launchers import SimpleLauncher
+from parsl.addresses import address_by_hostname
+from parsl.data_provider.files import File
+
+# Update to import config for your machine
+config = Config(
+    executors=[
+        FluxExecutor(
+            label="flux",
+            # Start Flux with srun and tell it there are 40 cores per node
+            launch_cmd='srun --mpi=pmi2 --tasks-per-node=1 -c2 ' + FluxExecutor.DEFAULT_LAUNCH_CMD,
+            provider=SlurmProvider(
+                channel=LocalChannel(),
+                nodes_per_block=3,
+                init_blocks=1,
+                partition='slurmpar',
+                account='',
+                walltime='00:10:00',
+                launcher=SimpleLauncher(),
+                worker_init='''
+# FluxExecutor ignores worker_init
+''',
+            ),
+        )
+    ],
+)
+
+
+import os
+
+parsl.load(config)
+
+remote = False
+shared_dir = './'
+
+# Print out resources that Flux sees after it starts
+@bash_app
+def resource_list():
+    return '''
+    flux resource list > resource_list.txt
+    '''
```

Comment:
I only see one reference to the resource_list function here and am unsure exactly what the use of this function is? It doesn't look like anything is being done within the function or returned from it. Do we still need to implement this? Unsure of the functionality here. Are we returning this docstring?

Reply:
The `flux resource list` command is a Flux command that asks Flux to print out the resources it thinks it has access to.  This is a simple test in the Parsl workflow that verifies that the resources we expect to be available are also seen by Flux.  The use of a returned docstring is the idiomatic way to implement Parsl workflow "Apps".  The contents of the `resource_list.txt` file contains the output of the command and the CI probably should actually check that the file is as expected.